### DISCUSSION TITLE
Migrate from xtext to langium

### DISCUSSION THREAD

**Discussion by harish-musarakallu** [eclipse-langium/langium#1144] [General]
Hello,
We have been using xtext for almost 10 years and it is integral part of our development. We are using it for multiple topics and one of them is for generating own grammar for defining the menu-tree (menus with submenus) containing upto 200 menu points along with language text.
From this validated menu-tree against the grammar, we also generate c code, generate documentation and also generate necessary files for text translations.

My question is how easy it is to migrate from eclipse based xtext infrastructure to langium. Also will there be any professional help could be arranged (as a business proposal) for us to ramp up the knowledge on this topic.

Look forward to the reply.
Best regards,
Harish

**Comment by msujew** [comment]
Hey @harish-musarakallu,

we actually wrote a generator fragment for translating Xtext based grammars to Langium. We wrote a blog post about it [here](https://www.typefox.io/blog/xtext-to-langium). It should take care of most of the heavy lifting - at least regarding the grammar migration. How easy it is to migrate the complete project heavily depends on how much you have built on top of the grammar. While a lot of services like LSP stuff and Scoping can be translated to TypeScript 1 to 1, some services, such as those related to workspace features work a bit differently in Langium than in Xtext.

If you need some professional support, [TypeFox is available for such requests](https://www.typefox.io/language-engineering/).
